api/src/main/java/io/opentelemetry/api/trace/TraceState.java

```diff
@@ -24,21 +21,40 @@
  *
  * <p>Value is opaque string up to 256 characters printable ASCII RFC0020 characters (i.e., the
  * range 0x20 to 0x7E) except comma , and =.
+ *
+ * <p>Implementations of this interface *must* be immutable and have well-defined value-based
+ * equals/hashCode implementations. If an implementation does not strictly conform to these
+ * requirements, behavior of the OpenTelemetry APIs and default SDK cannot be guaranteed.
+ *
+ * <p>Implementations of this interface that do not conform to the W3C specification risk
+ * incompatibility with W3C-compatible implementations.
+ *
+ * <p>For these reasons, it is strongly suggested that you use the implementation that is provided
+ * here via the {@link TraceStateBuilder}.
  */
 @Immutable
-@AutoValue
-public abstract class TraceState {
-  private static final TraceState DEFAULT = TraceState.builder().build();
+public interface TraceState {
+
+  TraceState DEFAULT = new TraceStateBuilder().build();
 
   /**
    * Returns the default {@code TraceState} with no entries.
    *
    * @return the default {@code TraceState}.
    */
-  public static TraceState getDefault() {
+  static TraceState getDefault() {
     return DEFAULT;
   }
 
+  /**
+   * Returns a {@code Builder} based on an empty {@code TraceState}.
+   *
+   * @return a {@code Builder} based on an empty {@code TraceState}.
+   */
+  static TraceStateBuilder builder() {
```

Comment:
I pondered a bit on whether this needs to return an interface too - but since it's static guess not

Reply:
@trask Actually can you check this - will we be able to do the interface interop if this doesn't return an interface?

Reply:
it's the `toBuilder()` method that hurts: https://github.com/open-telemetry/opentelemetry-java/pull/2134/files#r530758547

@jkwatson sorry to ask, would you mind turning these builders into interfaces?

Reply:
done. please take a look